netket/optimizer/sr/sr_onthefly_logic.py

```diff
@@ -184,7 +223,14 @@ def mat_vec(v, forward_fn, params, samples, diag_shift):
     diag_shift: a scalar diagonal shift
     """
 
-    res = Odagger_DeltaO_v(samples, params, v, forward_fn)
+    # TODO ideally we would make centered a static argument
+    # however fishing it out of the sr dataclass at jit is ugly
+    res = jax.lax.cond(
+        centered,
+        lambda args: DeltaOdagger_DeltaO_v(*args, forward_fn),
+        lambda args: Odagger_DeltaO_v(*args, forward_fn),
+        (samples, params, v),
+    )
```

Comment:
centered shouldn't be a conditional.
let's assume it's a static flag.
```struct.field(pytree_node=False)```

Reply:
> Also needs to edit `sr/api.py` to include the new flag

can't it just go into kwargs?